crypto/evp/pmeth_lib.c

```diff
@@ -264,6 +265,7 @@ EVP_PKEY_CTX *EVP_PKEY_CTX_dup(EVP_PKEY_CTX *pctx)
     if (pctx->pmeth->copy(rctx, pctx) > 0)
         return rctx;
 
+    rctx->pmeth = NULL;
```

Comment:
More details are in #2104 
EVP_PKEY_CTX_free would call ctx->pmeth->cleanup
but it either the copy method or the init method failed
so there is nothing to clean up, worse it may crash
in this scenario.